pallet/staking/src/lib.rs

```diff
@@ -735,215 +827,276 @@ pub mod pallet {
 			});
 		}
 
-		/// Add reward points to collators using their account id.
-		pub fn reward_by_ids(collators: &[(T::AccountId, RewardPoint)]) {
-			<RewardPoints<T>>::mutate(|(total, reward_map)| {
-				collators.iter().cloned().for_each(|(c, p)| {
-					*total += p;
+		/// Update the record of block production.
+		pub fn note_authors(authors: &[T::AccountId]) {
+			<AuthoredBlocksCount<T>>::mutate(|(sum, map)| {
+				authors.iter().cloned().for_each(|c| {
+					*sum += One::one();
 
-					reward_map.entry(c).and_modify(|p_| *p_ += p).or_insert(p);
+					map.entry(c).and_modify(|p_| *p_ += One::one()).or_insert(One::one());
 				});
 			});
 		}
 
-		// Power is a mixture of RING and KTON.
-		// - `total_ring_power = (amount / total_staked_ring) * HALF_POWER`
-		// - `total_kton_power = (amount / total_staked_kton) * HALF_POWER`
-		fn balance2power<P>(amount: Balance) -> Power
-		where
-			P: frame_support::StorageValue<Balance, Query = Balance>,
-		{
-			(Perquintill::from_rational(amount, P::get().max(1)) * 500_000_000_u128) as _
+		/// Calculate the power of the given account.
+		#[cfg(any(feature = "runtime-benchmarks", test))]
+		pub fn quick_power_of(who: &T::AccountId) -> Power {
```

Comment:
Why not:
```rs
#[cfg(any(feature = "runtime-benchmarks", test))]
pub fn quick_power_of(who: &T::AccountId) -> Power {
	let ring_pool = <RingPool<T>>::get();
	let kton_pool = <KtonPool<T>>::get();
	Self::power_of(who, ring_pool, kton_pool)
}
```

Reply:
Definitely. I didn't notice this before.